src/eos-metrics-instrumentation.c

```diff
@@ -276,62 +272,54 @@ get_user_id (GVariant *session_parameters, guint32 *user_id)
     return TRUE;
 }
 
-/*
- * Return TRUE if the given parameters of a SessionNew or SessionRemoved signal
- * correspond to a human session. If the given parameters don't correspond to a
- * human session, or this can't be determined, return FALSE.
- */
 static gboolean
-is_human_session (GVariant *session_parameters)
+is_human_session (const gchar *session_id)
 {
-    guint32 user_id;
-    if (!get_user_id (session_parameters, &user_id))
-      return FALSE;
-    return user_id >= MIN_HUMAN_USER_ID;
+    /* All normal user sessions start with digits -- greeter sessions
+     * start with 'c'.
+     */
+    return g_ascii_isdigit (session_id[0]);
 }
 
 /*
- * If the given parameters of a SessionNew or SessionRemoved signal correspond
- * to a human session, return TRUE and add the corresponding session_id to the
- * humanity_by_session_id set. Otherwise, return FALSE.
+ * If the given session_id corresponds to a human session not already in the set
+ * return TRUE and add the corresponding session_id to the humanity_by_session_id
+ * set. Otherwise, return FALSE.
  */
 static gboolean
-add_session_to_set (GVariant *session_parameters)
+add_session_to_set (const gchar *session_id)
 {
-    if (!is_human_session (session_parameters))
+    if (!is_human_session (session_id))
+      return FALSE;
+
+    gboolean in_set = (gboolean) GPOINTER_TO_UINT (
```

Comment:
If you were to break the line here, our style guide would expect the following vertical alignment:

```
gboolean in_set = (gboolean) GPOINTER_TO_UINT (
+                                              g_datalist_get_data (&humanity_by_session_id, session_id));
```

Since that doesn't do anything to reduce the line length, I recommend splitting this into multiple statements instead.